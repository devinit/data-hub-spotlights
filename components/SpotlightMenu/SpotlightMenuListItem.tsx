import classNames from 'classnames';
import React, { Children, FunctionComponent, useState, cloneElement, isValidElement } from 'react';

interface ComponentProps {
  title?: string;
  depth?: number;
  onView?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>, title?: string) => void;
}

const SpotlightMenuListItem: FunctionComponent<ComponentProps> = ({ title, children, onView: onViewProp, depth }) => {
  const [active, setActive] = useState(false);
  const toggleActive = (): void => {
    setActive(!active);
  };
  const onView = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>): void => {
    event.stopPropagation();
    if (onViewProp) {
      onViewProp(event, title);
    }
  };

  return (
    <li
      className={classNames({
        'countries-menu-list--has-children': children,
        'countries-menu-list__countries': !children,
        'js-profile-country-item': !children
      })}
    >
      <a
        className={classNames('countries-menu-list__item js-menu-item js-search-item', {
          active,
          'countries-menu-list__item--open': active && children,
          'countries-menu-list__item--parent-first': depth === 1,
          'countries-menu-list__item--parent-second': depth === 2,
          'countries-menu-list__item--parent-third': depth === 3,
          'countries-menu-list__item--parent-fourth': depth === 4,
          'countries-menu-list__item--parent-fifth': depth === 5,
          'countries-menu-list__item--parent-sixth': depth === 6
        })}
        onClick={toggleActive}
      >
        {title}
        <style jsx>{`
          :before {
            display: ${children ? 'inline-block' : 'none'};
          }
        `}</style>
      </a>
      <a
        className="countries-menu__profile countries-menu__link js-profile-item"
        onClick={onView}
        title={`View ${title}`}
      >
        View
      </a>
      {Children.map(children, child => isValidElement(child) && cloneElement(child, { active }))}
    </li>
  );
};

SpotlightMenuListItem.defaultProps = { depth: 1 };

export { SpotlightMenuListItem };
