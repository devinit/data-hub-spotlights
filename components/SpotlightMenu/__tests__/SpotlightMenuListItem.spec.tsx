/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SpotlightMenuListItem } from '../SpotlightMenuListItem';

describe('SpotlightMenuListItem', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<SpotlightMenuListItem title={'Busia'} />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    const renderer = TestRenderer.create(
      <SpotlightMenuListItem title={'Palisa'}>
        <div>Testing</div>
      </SpotlightMenuListItem>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
