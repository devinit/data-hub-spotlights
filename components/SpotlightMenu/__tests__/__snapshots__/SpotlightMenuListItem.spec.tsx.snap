// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotlightMenuListItem renders children correctly 1`] = `
<li
  className="countries-menu-list--has-children"
>
  <a
    className="countries-menu-list__item js-menu-item js-search-item countries-menu-list__item--parent-first"
    onClick={[Function]}
  >
    palisa
    <style
      jsx={true}
    >
      
          :before {
            display: inline-block;
          }
        
    </style>
  </a>
  <a
    className="countries-menu__profile countries-menu__link js-profile-item"
    onClick={[Function]}
    title="View Palisa"
  >
    View
    <style
      jsx={true}
    >
      
          
        
    </style>
  </a>
  <div
    active={false}
  >
    Testing
  </div>
</li>
`;

exports[`SpotlightMenuListItem renders correctly 1`] = `
<li
  className="countries-menu-list__countries js-profile-country-item"
>
  <a
    className="countries-menu-list__item js-menu-item js-search-item countries-menu-list__item--parent-first"
    onClick={[Function]}
  >
    busia
    <style
      jsx={true}
    >
      
          :before {
            display: none;
          }
        
    </style>
  </a>
  <a
    className="countries-menu__profile countries-menu__link js-profile-item"
    onClick={[Function]}
    title="View Busia"
  >
    View
    <style
      jsx={true}
    >
      
          
        
    </style>
  </a>
</li>
`;
